codegenerator/cli/templates/static/codegen/src/eventFetching/ChainFetcher.res

```diff
@@ -355,17 +355,13 @@ let updateFetchState = (
 }
 
 /**
-Gets the next query either with a to block of the current height if it is the root node.
-Or with a toBlock of the nextRegistered latestBlockNumber to catch up and merge with the next regisetered.
+Gets the next queries for all most behind partitions not exceeding queue size
 
 Applies any event filters found in the chain fetcher
-
-Errors if nextRegistered dynamic contract has a lower latestFetchedBlock than the current as this would be
-an invalid state.
 */
-let getNextQuery = (self: t, ~maxPerChainQueueSize) => {
-  self.fetchState->PartitionedFetchState.getNextQueriesOrThrow(
-    ~currentBlockHeight=self.currentBlockHeight,
+let getNextQueries = (self: t, ~maxPerChainQueueSize) => {
+  self.fetchState->PartitionedFetchState.getNextQueries(
```

Comment:
This doesn't interact with currentBlockHeight anymore. I think it made things easier.